kornia/augmentation/augmentations.py

```diff
@@ -1,21 +1,35 @@
-from typing import Callable, Tuple, List, Union, cast
+from typing import Callable, Tuple, Union, List, Optional, Dict, cast
 
 import torch
 import torch.nn as nn
-from torch.distributions import Uniform
 
-from kornia.geometry.transform.flips import hflip, vflip
-from kornia.color.adjust import AdjustBrightness, AdjustContrast, AdjustSaturation, AdjustHue
-from kornia.color.gray import rgb_to_grayscale
-from kornia.geometry import pi
+import kornia.augmentation.functional as F
+import kornia.augmentation.param_gen as pg
 
 UnionType = Union[torch.Tensor, Tuple[torch.Tensor, torch.Tensor]]
 FloatUnionType = Union[torch.Tensor, float, Tuple[float, float], List[float]]
 
 
-class RandomFlip(nn.Module):
+class AugmentationBase(nn.Module):
+    def __init__(self, return_transform: bool = False) -> None:
+        super(AugmentationBase, self).__init__()
+        self.return_transform = return_transform
+
+    def __repr__(self) -> str:
+        repr = f"(return_transform={self.return_transform})"
+        return self.__class__.__name__ + repr
+
+    def forward(self, input: UnionType) -> UnionType:  # type: ignore
```

Comment:
This is definitely a good suggestion to have a better abstraction of those functions.

For param generator part, I am thinking of something like:
```python
class ColorJitterParamDictGenerator():
      def random(a, b, c, d) -> dict:
             # 1. Generate random params
             # 2. Put params together
            return compose(_a, _b, _c, _d)
      def compose(a, b, c, d) -> dict:
            # This will be used for static params.
            return param_dict
      def validate(param_dict) -> bool:
            return valid_or_not
```
Also, I think the param generator should not do any ```.to(device).to(dtype)``` operations. The generator generates valid parameters, which is not device sensitive. Plus, in multi-gpus settings, one set of parameters may be used in different GPUs, which means its totally depending on the device and dtype of the batch images.